near-sdk/src/json_types/hash.rs

```diff
@@ -60,6 +59,14 @@ impl TryFrom<&str> for Base58CryptoHash {
     type Error = Box<dyn std::error::Error>;
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
+        Self::from_str(value)
+    }
+}
+
+impl FromStr for Base58CryptoHash {
```

Comment:
As an alternative, I sometimes like to write similar impls as 

```suggestion
impl std::str::FromStr for Base58CryptoHash {
```

that way, you avoid importing the trait, so you explicitly show that you are *not* using the trait, and only implement it.

Reply:
ah, that's actually a good practice, I didn't think of that. I'll change so it's future proofed!

Reply:
actually, it is used in this file (in the deserialize)